### PR TITLE
Open selected file when launching Android Studio

### PR DESCRIPTION
**Pull request by stevemessick**
@pq 
If the "Open in Android Studio" action is invoked on a file (not directory) in IntelliJ then Android Studio will open that file for editing when it is launched.

Have not looked at AS -> AS yet.

Tried to do IJ -> Xcode but that didn't work.

### CODE REVIEW

File: src/io/flutter/actions/OpenInAndroidStudioAction.java [hunk 1]
```diff
@@ -30,11 +30,14 @@
 
 public class OpenInAndroidStudioAction extends AnAction {
 
-  private static void openFileInStudio(@NotNull VirtualFile file, @NotNull String androidStudioPath) {
+  private static void openFileInStudio(@NotNull VirtualFile file, @NotNull String androidStudioPath, @Nullable String sourceFile) {
```
Comment (pq) [inline src/io/flutter/actions/OpenInAndroidStudioAction.java]: For understandability, I wonder if we couldn't change `file` to `projectFile`?

File: src/io/flutter/actions/OpenInAndroidStudioAction.java [hunk 2]
```diff
@@ -30,11 +30,14 @@
...
       final GeneralCommandLine cmd;
       if (SystemInfo.isMac) {
-        cmd = new GeneralCommandLine().withExePath("open").withParameters("-a", androidStudioPath, file.getPath());
+        cmd = new GeneralCommandLine().withExePath("open").withParameters("-a", androidStudioPath, "--args", file.getPath());
+        if (sourceFile != null) {
+          cmd.addParameter(sourceFile);
+        }
       }
       else {
         cmd = new GeneralCommandLine().withExePath(androidStudioPath).withParameters(file.getPath());
```
Comment (pq) [inline src/io/flutter/actions/OpenInAndroidStudioAction.java]: Is this known not to work for Windows or is it just unexplored?  (If it's the latter, maybe a TODO?)

File: src/io/flutter/actions/OpenInAndroidStudioAction.java [hunk 3]
```diff
@@ -219,6 +222,8 @@ public void actionPerformed(AnActionEvent e) {
       return;
     }
 
-    openFileInStudio(projectFile, androidStudioPath);
+    VirtualFile file = e.getData(CommonDataKeys.VIRTUAL_FILE);
```
Comment (pq) [inline src/io/flutter/actions/OpenInAndroidStudioAction.java]: Add final?

File: src/io/flutter/actions/OpenInAndroidStudioAction.java [hunk 4]
```diff
@@ -219,6 +222,8 @@ public void actionPerformed(AnActionEvent e) {
...
+    String sourceFile = file == null ? null : file.isDirectory() ? null : file.getPath();
```
Comment (pq) [inline src/io/flutter/actions/OpenInAndroidStudioAction.java]: Add final?

File: src/io/flutter/actions/OpenInAndroidStudioAction.java (hunk 1 above)
Comment (stevemessick) [reply src/io/flutter/actions/OpenInAndroidStudioAction.java]: Sure.

File: src/io/flutter/actions/OpenInAndroidStudioAction.java (hunk 2 above)
Comment (stevemessick) [reply src/io/flutter/actions/OpenInAndroidStudioAction.java]: It is known to work for Mac. I added a TODO.

File: src/io/flutter/actions/OpenInAndroidStudioAction.java (hunk 4 above)
Comment (stevemessick) [reply src/io/flutter/actions/OpenInAndroidStudioAction.java]: Let's churn

**Comment by pq** [comment]
@stevemessick: With this landed,  I'll have some analytics integrated shortly!

**Pull request merged**